.github/workflows/publish-canary.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish canary release
+on:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  tag-and-publish-to-npm:
+    runs-on: ubuntu-latest
+
+    # Don't run on regular releases
+    if: "!startsWith(github.event.head_commit.message, 'release: ')"
+    steps:
+      - uses: actions/checkout@v2.0.0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "18.16.1"
```

Comment:
Not directly related to this PR, but notice we don't specify the node version anyway (in `.npmrc`; or in a `.node-version` or `package.json`) unless I missed it. I think we probably should do (one of those), to ensure that local dev builds are running with the same version as CI.